src/app/modules/verification/verification.middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { RequestHandler } from 'express'
+import { StatusCodes } from 'http-status-codes'
+
+import featureManager, { FeatureNames } from '../../../config/feature-manager'
+
+import * as verification from './verification.controller'
+
+interface IVerifiedFieldsMiddleware {
+  createTransaction: RequestHandler
+  resetFieldInTransaction: RequestHandler<{ transactionId: string }>
+  getNewOtp: RequestHandler<{ transactionId: string }>
+  verifyOtp: RequestHandler<{ transactionId: string }>
+}
+
+const verificationMiddlewareFactory = ({
+  isEnabled,
+}: {
+  isEnabled: boolean
+}): IVerifiedFieldsMiddleware => {
+  if (isEnabled) {
+    return {
+      createTransaction: verification.createTransaction,
+      resetFieldInTransaction: verification.resetFieldInTransaction,
+      getNewOtp: verification.getNewOtp,
+      verifyOtp: verification.verifyOtp,
+    }
+  } else {
+    const errMsg = 'Verified fields feature is not enabled'
+    return {
+      createTransaction: (req, res) =>
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ message: errMsg }),
```

Comment:
do we not use `MissingFeatureError` anymore?

Reply:
we do! it's just that these are the middleware functions which haven't been migrated yet. every subsequent PR in this series will remove one of these functions and replace it with one in `verification.controller.ts`